src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -841,8 +841,17 @@ HRESULT CProfilerManager::SetEventMask(DWORD dwEventMask)
     }
     else
     {
-        CLogging::LogError(_T("SetEventMask can only be called during initialize of Host or InstrumentationMethod"));
-        return E_FAIL;
+        // Prevent immutable flags from being set.
+        if ((dwEventMask & COR_PRF_MONITOR_IMMUTABLE) !=
```

Comment:
Make this "else if" and flip the condition so that true is valid (and sets the flags) and the else block is the failure.